library/src/main/java/org/mustangproject/ZUGFeRD/ZUGFeRDInvoiceImporter.java

```diff
@@ -646,89 +646,47 @@ public Invoice extractInto(Invoice zpp) throws XPathExpressionException, ParseEx
-						} else if (chargeChildName.equals("CategoryTradeTax")) {
-							NodeList taxChilds = chargeNodeChilds.item(chargeChildIndex).getChildNodes();
-							for (int taxChildIndex = 0; taxChildIndex < taxChilds.getLength(); taxChildIndex++) {
-								String taxItemName = taxChilds.item(taxChildIndex).getLocalName();
-								if ((taxItemName != null) && (taxItemName.equals("RateApplicablePercent") || taxItemName.equals("ApplicablePercent"))) {
-									taxPercent = XMLTools.trimOrNull(taxChilds.item(taxChildIndex));
-								}
-							}
-						}
+				xpr = xpath.compile(".//*[local-name()=\"SpecifiedTradeAllowanceCharge\"]");
+				NodeList chargeNodes = (NodeList) xpr.evaluate(currentItemNode, XPathConstants.NODESET);
+				for (int k = 0; k < chargeNodes.getLength(); k++) {
+					NodeList chargeNodeChilds = chargeNodes.item(k).getChildNodes();
+					Charge c = parseCharge(chargeNodeChilds, false);
+					if (c instanceof Allowance) {
+						it.addTotalAllowance(c);
+					} else {
+						it.addCharge(c);
 					}
 				}
+			}
+		}
 
-				if (isCharge) {
-					Charge c = new Charge(new BigDecimal(chargeAmount));
-					if (reason != null) {
-						c.setReason(reason);
-					}
-					if (reasonCode != null) {
-						c.setReasonCode(reasonCode);
-					}
-					if (taxPercent != null) {
-						c.setTaxPercent(new BigDecimal(taxPercent));
-					}
-					zpp.addCharge(c);
-				} else {
-					Allowance a = new Allowance(new BigDecimal(chargeAmount));
-					if (reason != null) {
-						a.setReason(reason);
-					}
-					if (reasonCode != null) {
-						a.setReasonCode(reasonCode);
-					}
-					if (taxPercent != null) {
-						a.setTaxPercent(new BigDecimal(taxPercent));
-					}
-					zpp.addAllowance(a);
-				}
-
+		// now handling base64 encoded attachments AttachmentBinaryObject=CII, EmbeddedDocumentBinaryObject=UBL
+		xpr = xpath.compile("//*[local-name()=\"AttachmentBinaryObject\"]|//*[local-name()=\"EmbeddedDocumentBinaryObject\"]");
+		NodeList attachmentNodes = (NodeList) xpr.evaluate(getDocument(), XPathConstants.NODESET);
+		for (int i = 0; i < attachmentNodes.getLength(); i++) {
+			FileAttachment fa = new FileAttachment(attachmentNodes.item(i).getAttributes().getNamedItem("filename").getNodeValue(), attachmentNodes.item(i).getAttributes().getNamedItem("mimeCode").getNodeValue(), "Data", Base64.getDecoder().decode(XMLTools.trimOrNull(attachmentNodes.item(i))));
+			fileAttachments.add(fa);
+			// filename = "Aufmass.png" mimeCode = "image/png"
+			//EmbeddedDocumentBinaryObject cbc:EmbeddedDocumentBinaryObject mimeCode="image/png" filename="Aufmass.png"
+		}
+
+		// item level charges+allowances are not yet handled but a lower item price will
+		// be read,
+		// so the invoice remains arithmetically correct
+		// -> parse document level charges+allowances
+		xpr = xpath.compile("//*[local-name()=\"ApplicableHeaderTradeSettlement\"]//*[local-name()=\"SpecifiedTradeAllowanceCharge\"]");
```

Comment:
Added ApplicableHeaderTradeSettlement because we only want to have header level charges here